server/catgenome/src/main/java/com/epam/catgenome/manager/FeatureIndexManager.java

```diff
@@ -712,6 +714,99 @@ public void processGeneFile(GeneFile geneFile, final Map<String, Chromosome> chr
         }
     }
 
+    public NggbIntervalTreeMap<List<Gene>> loadGenesIntervalMap(final List<GeneFile> geneFiles, final int start,
+                                                                final int end, final Chromosome chromosome) {
+        final NggbIntervalTreeMap<List<Gene>> genesRangeMap = new NggbIntervalTreeMap<>();
+        try {
+            IndexSearchResult<FeatureIndexEntry> searchResult = featureIndexDao.searchFeaturesInInterval(geneFiles,
+                    start, end, chromosome);
+            searchResult.getEntries().stream()
+                    .filter(f -> f.getFeatureType() == FeatureType.EXON || f.getFeatureType() == FeatureType.GENE)
+                    .map(f -> {
+                        Gene gene = new Gene();
+                        gene.setFeature(f.getFeatureType().name());
+                        gene.setStartIndex(f.getStartIndex());
+                        gene.setEndIndex(f.getEndIndex());
+                        gene.setGroupId(f.getFeatureId());
+                        gene.setFeatureName(f.getFeatureName().toUpperCase());
+                        return gene;
+                    })
+                    .forEach(g -> {
+                        Interval interval = new Interval(chromosome.getName(), g.getStartIndex(), g.getEndIndex());
+                        genesRangeMap.putIfAbsent(interval, new ArrayList<>());
+                        genesRangeMap.get(interval).add(g);
+                    });
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        genesRangeMap.setMaxEndIndex(start);
+        genesRangeMap.setMinStartIndex(end);
+        return genesRangeMap;
+    }
+
+    /**
+     * Fetch gene IDs of genes, affected by variation. The variation is specified by it's start and end indexes
+     *
+     * @param intervalMap represents a batch loaded genes form gene file
+     * @param start       a start index of the variation
+     * @param end         an end index of the variation
+     * @param chromosome  a {@code Chromosome}
+     * @return a {@code Set} of IDs of genes, affected by the variation
+     */
+    public static Set<GeneInfo> fetchGeneIdsFromBatch(final NggbIntervalTreeMap<List<Gene>> intervalMap,
```

Comment:
Let's move new static methods like `fetchGeneIdsFromBatch` to a separate util class e.g. `FeatureIndexUtils`